public/app/features/logs/components/LogRowContext.tsx

```diff
@@ -61,26 +65,60 @@ const getLogRowContextStyles = (theme: GrafanaTheme2, wrapLogMessage?: boolean)
       box-shadow: 0 0 10px ${theme.v1.palette.black};
       border: 1px solid ${theme.colors.background.secondary};
       border-radius: ${theme.shape.borderRadius(2)};
-      width: 100%;
+      font-family: ${theme.typography.fontFamily};
     `,
     header: css`
       height: ${headerHeight}px;
       padding: 0 10px;
       display: flex;
       align-items: center;
+      background: ${theme.colors.background.canvas};
+    `,
+    top: css`
+      border-radius: 0 0 ${theme.shape.borderRadius(2)} ${theme.shape.borderRadius(2)};
+      box-shadow: 0 0 10px ${theme.v1.palette.black};
+      clip-path: inset(0px -10px -10px -10px);
+    `,
+    title: css`
+      position: absolute;
+      top: -${contextHeight + headerHeight}px;
+      z-index: ${theme.zIndex.modal};
+      height: ${headerHeight}px;
       background: ${theme.colors.background.secondary};
+      border: 1px solid ${theme.colors.background.secondary};
+      border-radius: ${theme.shape.borderRadius(2)} ${theme.shape.borderRadius(2)} 0 0;
+      box-shadow: 0 0 10px ${theme.v1.palette.black};
+      clip-path: inset(-10px -10px 0px -10px);
```

Comment:
I am using box-shadows in combination with `clip-path` to hide shadows on the relevant sides.